File: view/src/main/java/by/rekuts/travelagency/view/controller/GlobalDefaultExceptionHandler.java
package by.rekuts.travelagency.view.controller;

import org.springframework.core.annotation.AnnotationUtils;
import org.springframework.web.bind.annotation.ControllerAdvice;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.ResponseStatus;
import org.springframework.web.servlet.ModelAndView;

import javax.servlet.http.HttpServletRequest;

@ControllerAdvice
class GlobalDefaultExceptionHandler {
    @ExceptionHandler(value = Exception.class)
    public ModelAndView defaultErrorHandler(HttpServletRequest req, Exception e) throws Exception {
        if (AnnotationUtils.findAnnotation
                (e.getClass(), ResponseStatus.class) != null)
            throw e;
        ModelAndView mav = new ModelAndView();
        mav.addObject("name", e.getClass().getName());
        mav.addObject("message", e.getLocalizedMessage());
        mav.addObject("stackTrace", e.getStackTrace());
        mav.addObject("url", req.getRequestURL());
        mav.setViewName("error");
        return mav;
    }
}

File: view/src/main/scala/by/rekuts/travelagency/view/controller/GlobalDefaultExceptionHandler.scala
package by.rekuts.travelagency.view.controller

import javax.servlet.http.HttpServletRequest
import org.springframework.core.annotation.AnnotationUtils
import org.springframework.web.bind.annotation.{ControllerAdvice, ExceptionHandler, ResponseStatus}
import org.springframework.web.servlet.ModelAndView

@ControllerAdvice
class GlobalDefaultExceptionHandler {
  @ExceptionHandler(value = Array(classOf[Exception]))
  @throws[Exception]
  def defaultErrorHandler(req: HttpServletRequest, e: Exception): ModelAndView = {
    if (AnnotationUtils.findAnnotation(e.getClass, classOf[ResponseStatus]) != null) throw e
    val mav = new ModelAndView
    mav.addObject("name", e.getClass.getName)
    mav.addObject("message", e.getLocalizedMessage)
    mav.addObject("stackTrace", e.getStackTrace)
    mav.addObject("url", req.getRequestURL)
    mav.setViewName("error")
    mav
  }
}
